cluster/Vagrantfile

```diff
@@ -4,32 +4,49 @@ require 'rubygems'
 require 'fileutils'
 require 'yaml'
 
-DEFAULT_K8S_VERSION = 'v1.4.7'.freeze
+DEFAULT_K8S_VERSION = 'v1.6.4'.freeze
+
+# Guest OS versions for testing
 UBUNTU = 'ubuntu'.freeze
 CENTOS = 'centos'.freeze
 RHEL = 'rhel7'.freeze
 
+HOST_SHARED_FOLDER = './export/'.freeze
+# Different orchestration platforms we support
+ORC_LEGACY_SWARM = 'docker'.freeze
+ORC_SWARM = 'swarm'.freeze
+ORC_KUBEADM = 'kubeadm'.freeze
+ORC_KUBEADM_IDX = 2
+
+# Global vars
 token = 'd900e1.8a392798f13b33a4'
 node_os = ENV['CONTIV_NODE_OS'] || CENTOS
+orc = ENV['CONTIV_ORC'] || ORC_LEGACY_SWARM
+k8s_ver = ENV['CONTIV_K8S_VERSION'] || DEFAULT_K8S_VERSION
+orc_path = k8s_ver.include?('1.5.') || k8s_ver.include?('1.4.') ? 'k8s1.4/' : 'k8s1.6/'
+swarm_path = 'docker17/'
+orchestrators = [ORC_LEGACY_SWARM, ORC_SWARM, ORC_KUBEADM]
 
 # method to create an etc_hosts file based on the cluster info
-def create_etc_hosts(num_nodes, base_ip, start)
-  master_ip = base_ip + start.to_s
+def create_etc_hosts(node_names, node_ips, o)
+  master_ip = node_ips[0]
   hosts = "127.0.0.1   localhost\n"
   hosts << "#{master_ip}   netmaster\n"
-  hosts = (0..num_nodes).inject(hosts) { |acc, elem| acc << base_ip + "#{elem + start} contiv-node#{elem + 1} \n" }
 
-  etc_file = (ENV['VAGRANT_CWD'] || '.') + '/.etc_hosts'
+  node_names.zip(node_ips).each do |node, ip|
+    hosts = hosts + node + ' ' + ip + '\n'
```

Comment:
I would change this to

```rb
hosts << "#{node} #{ip}\n"
```

to be consistent with the string appending above it